### PR TITLE
Fix hash family functions signatures

### PR DESCRIPTION
**Pull request by norkunas**

### CODE REVIEW

**Comment by ondrejmirtes** [comment]
Perhaps you can modify more similar functions like `md5`, `hash`, etc.? Thanks.

File: resources/functionMap.php [hunk 1]
```diff
@@ -8077,7 +8077,7 @@
 'openssl_csr_sign' => ['resource|false', 'csr'=>'string|resource', 'x509'=>'string|resource|null', 'priv_key'=>'string|resource|array', 'days'=>'int', 'config_args='=>'array', 'serial='=>'int'],
 'openssl_decrypt' => ['string|false', 'data'=>'string', 'method'=>'string', 'key'=>'string', 'options='=>'int', 'iv='=>'string', 'tag='=>'string', 'aad='=>'string'],
 'openssl_dh_compute_key' => ['string|false', 'pub_key'=>'string', 'dh_key'=>'resource'],
-'openssl_digest' => ['string|false', 'data'=>'string', 'method'=>'string', 'raw_output='=>'bool'],
+'openssl_digest' => ['string|false', 'data'=>'string', 'method'=>'string', 'binary'=>'bool'],
```
Comment (staabm) [inline resources/functionMap.php:8080]:
```suggestion
'openssl_digest' => ['non-empty-string|false', 'data'=>'string', 'method'=>'string', 'binary'=>'bool'],
```

**Comment by norkunas** [comment]
I just don't understand why there in `'raw_output='=>'bool'` was an equal sign as suffix, there are also more places like this but not all signatures follows that. What's the purpose of it?

**Comment by staabm** [comment]
I think it would also work for `crc32` and `crypt`

File: resources/functionMap_php80delta.php [hunk 2]
```diff
@@ -45,7 +45,7 @@
 		'get_resource_id' => ['int', 'res'=>'resource'],
 		'gmdate' => ['string', 'format'=>'string', 'timestamp='=>'int'],
 		'gmmktime' => ['int|false', 'hour'=>'int', 'minute='=>'int', 'second='=>'int', 'month='=>'int', 'day='=>'int', 'year='=>'int'],
-		'hash_hkdf' => ['string', 'algo'=>'string', 'ikm'=>'string', 'length='=>'int', 'info='=>'string', 'salt='=>'string'],
+		'hash_hmac' => ['non-empty-string', 'algo'=>'string', 'data'=>'string', 'key'=>'string', 'binary='=>'bool'],
```
Comment (staabm) [inline resources/functionMap_php80delta.php:48]: did you intentionally replace `hash_hkdf` with `hash_hmac`?

Comment (norkunas) [reply resources/functionMap_php80delta.php:48]: No, sorry :D

**Comment by staabm** [comment]
note: there is some overlap with https://github.com/phpstan/phpstan-src/pull/822/files

**Comment by norkunas** [comment]
> note: there is some overlap with #822

Then I think I could close this one as that PR is more complete :)

**Comment by staabm** [comment]
> I just don't understand why there in `'raw_output='=>'bool'` was an equal sign as suffix, there are also more places like this but not all signatures follows that. What's the purpose of it?

the format of the config file is described at the top level comment, see https://github.com/phpstan/phpstan-src/blob/446a0c8d95473b8830fe7c781d755ac0946ce411/resources/functionMap.php#L38

**Comment by norkunas** [comment]
> > I just don't understand why there in `'raw_output='=>'bool'` was an equal sign as suffix, there are also more places like this but not all signatures follows that. What's the purpose of it?
> 
> @norkunas the format of the config file is described at the top level comment, see
> 
> https://github.com/phpstan/phpstan-src/blob/446a0c8d95473b8830fe7c781d755ac0946ce411/resources/functionMap.php#L38

Thank you

**Pull request closed without merging**